packages/vite/src/node/plugins/manifest.ts

```diff
@@ -32,13 +32,52 @@ export function manifestPlugin(config: ResolvedConfig): Plugin {
     return perEnvironmentPlugin('native:manifest', (environment) => {
       if (!environment.config.build.manifest) return false
 
-      return nativeManifestPlugin({
-        root: environment.config.root,
-        outPath:
-          environment.config.build.manifest === true
-            ? '.vite/manifest.json'
-            : environment.config.build.manifest,
-      })
+      const root = environment.config.root
+      const outPath =
+        environment.config.build.manifest === true
+          ? '.vite/manifest.json'
+          : environment.config.build.manifest
+
+      function getChunkName(chunk: OutputChunk) {
+        if (chunk.facadeModuleId) {
+          return normalizePath(
+            path.relative(root, chunk.facadeModuleId),
+          ).replace(/\0/g, '')
+        }
+        return `_${path.basename(chunk.fileName)}`
+      }
+
+      return [
+        nativeManifestPlugin({ root, outPath }),
+        {
+          name: 'native:manifest-compatible',
+          generateBundle(_, bundle) {
+            const asset = bundle[outPath]
+            if (asset.type === 'asset') {
+              let manifest = null
+              for (const chunk of Object.values(bundle)) {
+                if (chunk.type !== 'chunk') continue
+                const importedCss = chunk.viteMetadata?.importedCss
+                const importedAssets = chunk.viteMetadata?.importedAssets
+                if (!importedCss?.size && !importedAssets?.size) continue
+                if (!manifest) manifest = JSON.parse(asset.source.toString())
```

Comment:
```suggestion
              let manifest: Manifest | undefined
              for (const chunk of Object.values(bundle)) {
                if (chunk.type !== 'chunk') continue
                const importedCss = chunk.viteMetadata?.importedCss
                const importedAssets = chunk.viteMetadata?.importedAssets
                if (!importedCss?.size && !importedAssets?.size) continue
                manifest ??= JSON.parse(asset.source.toString()) as Manifest
```
Let's make the `manifest` variable a bit more type-safe.